backend/routes/volunteer.js

```diff
@@ -65,21 +66,42 @@ router.post(
         return res.status(400).json({ errors: errors.array() });
       }
 
-      const volunteer = await Volunteer.create({
-        firstName: req.body.firstName,
-        lastName: req.body.lastName,
+      // add this later
+      let volunteer = await Volunteer.findOne({
         email: req.body.email,
-        phone: req.body.phone,
-        // initialize signedWaiver to false when Volunteer is created
-        signedWaiver: false,
+        firstName: req.body.firstName,
       });
+      // check if volunteer with req email exists
+      if (volunteer === null) {
+        // create new volunteer with same email but different name
+        volunteer = await Volunteer.create({
+          firstName: req.body.firstName,
+          lastName: req.body.lastName,
+          email: req.body.email,
+          phone: req.body.phone,
+          // initialize signedWaiver to false when new Volunteer is created
+          signedWaiver: false,
+        });
+        volunteer.save();
+      }
 
       // update event with volunteer
       const { eventID } = req.body;
-      await Event.findByIdAndUpdate(eventID, {
-        $push: { volunteers: volunteer._id },
+      const event = await Event.findOne({
+        _id: mongoose.Types.ObjectId(eventID),
       });
-
+      if (!event.volunteers.includes(volunteer._id)) {
+        await event.updateOne({
+          // addToSet checks for duplicates
+          $addToSet: { volunteers: volunteer._id },
+        });
+      } else {
+        res.json(volunteer);
+        // eslint-disable-next-line no-throw-literal
+        // throw {
+        //   message: "Volunteer array already contains user for this event",
+        // };
```

Comment:
Right now the backend crashes when a user signs up for an event they're already signed up for. Could you handle this edge case?